D 1.4	05/05/24 14:18:14 wangd	5 4	00009/00002/01375
fixed CR6261221: view page source causes mozilla crash on S10 b74L2a

diff -urp -x '*~' -x '*.orig' src/xftcore.c src/xftcore.c
--- src/xftcore.c	2006-12-07 02:17:19.000000000 -0800
+++ src/xftcore.c	2007-06-11 17:36:56.256417000 -0700
@@ -1254,8 +1254,15 @@ XftGlyphFontSpecCore (XftDraw			*draw,
 	g_y2 = g_y1 + gi.height;
 	if (i)
 	{
-	    if (g_x1 < x1)
-		x1 = g_x1;
+	    if (g_x1 < x1) {
+                if(g_x1 < 0 ) {
+                    /* do nothing if the given glyphs are out of range */
+                    short t = glyphs[i-1].font->max_advance_width + glyphs[i-1].x;
+                    if(t < 0 && glyphs[i-1].x > 0)
+                        goto bail1;
+                }
+                x1 = g_x1;
+            }
 	    if (g_y1 < y1)
 		y1 = g_y1;
 	    if (g_x2 > x2)
